### DISCUSSION TITLE
Unify input syntax

### DISCUSSION THREAD

**Discussion by jkosinski** [KosinskiLab/AlphaPulldown#455] [Ideas]
Just parking it here for future discussion - we have now two syntaxes, new for snakemake mode and legacy for script-based mode, and as a user of both I already find it confusing. We should consider adding the new syntax also for the scripts mode.

**Comment by DimaMolod** [comment]
> Just parking it here for future discussion - we have now two syntaxes, new for snakemake mode and legacy for script-based mode, and as a user of both I already find it confusing. We should consider adding the new syntax also for the scripts mode.

Just a comment: it's actually different syntax for `run_multimer_jobs.py` and `run_structure_prediction.py`, has nothing to do with snakemake. The newer syntax is better, but we keep the original one in `run_multimer_jobs.py` for back compatibility reasons.

**Comment by jkosinski** [comment]
It might be that we will need to keep `run_multimer_jobs.py` or implement equivalent job management in run_structure_prediction.py as for single or few jobs snakemake pipeline feels adding too much overhead (in my cases docker pulls and symlink jobs to reuse features taking more than actual prediction, plus all those extra files and storage that for now seems worthwhile). This could be probably as sth as simple as just supporting custom mode (so raw new syntax) in run_structure_prediction.py  but just adding a possibility to run over all lines in an input file.

**Reply by DimaMolod** [reply]
or we can just re-use [this function](https://github.com/KosinskiLab/AlphaPulldown/blob/main/alphapulldown/utils/create_combinations.py#L18) if input is a file in `run_structure_prediction.py`, similarly to `run_multimer_jobs.py`

**Reply by DimaMolod** [reply]
@maurerv but this function seems to be duplicated and different between [snakemake](https://github.com/KosinskiLab/AlphaPulldownSnakemake/blob/main/workflow/rules/common.smk#L20) and [alphapulldown](https://github.com/KosinskiLab/AlphaPulldown/blob/main/alphapulldown/utils/create_combinations.py#L18)?!
